template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/IdentityService/Controllers/IdentityServiceController.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity.Service;
+using Microsoft.AspNetCore.Identity.Service.IntegratedWebClient;
+using Microsoft.AspNetCore.Identity.Service.Mvc;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Company.WebApplication1.Identity.Models;
+
+namespace Company.WebApplication1.Identity.Controllers
+{
+    [Area(IdentityServiceConstants.Tenant)]
+    [Route("tfp/[area]/" + IdentityServiceConstants.DefaultPolicy + "/oauth2/v" + IdentityServiceConstants.Version + "/[action]")]
+    public class IdentityServiceController : Controller
+    {
+        private readonly IOptions<IdentityServiceOptions> _options;
+        private readonly ITokenManager _tokenManager;
+        private readonly SessionManager<ApplicationUser, IdentityServiceApplication> _sessionManager;
+        private readonly IAuthorizationResponseFactory _authorizationResponseFactory;
+        private readonly ITokenResponseFactory _tokenResponseFactory;
+
+        public IdentityServiceController(
+            IOptions<IdentityServiceOptions> options,
+            ITokenManager tokenManager,
+            SessionManager<ApplicationUser, IdentityServiceApplication> sessionManager,
+            IAuthorizationResponseFactory authorizationResponseFactory,
+            ITokenResponseFactory tokenResponseFactory)
+        {
+            _options = options;
+            _tokenManager = tokenManager;
+            _sessionManager = sessionManager;
+            _authorizationResponseFactory = authorizationResponseFactory;
+            _tokenResponseFactory = tokenResponseFactory;
+        }
+
+        [AcceptVerbs("GET", "POST")]
+        [ActionName("Authorize")]
+        public async Task<IActionResult> AuthorizeAsync(
+            [EnableIntegratedWebClient, ModelBinder(typeof(AuthorizationRequestModelBinder))] AuthorizationRequest authorization)
+        {
+            if (!authorization.IsValid)
+            {
+                return this.InvalidAuthorization(authorization.Error);
```

Comment:
Using extension methods on `Controller` doesn't work with POCO controllers. What exactly are the extension methods looking for? Could they be written as `static` methods with the right parameters (not the whole `Controller`)?